Custom tags with hyphen in *.jsx files crash JavaScript analysis

**I want to report a bug.**

**SonarJS version:** 3.3
**SonarQube version:** 6.3

**Reproducer** 
```
render() {
         return <g>
             <if condition={ this.src }>
                 <img { ...this.undeclaredAttributes() }
                     class= { this._thumbnailClasses }
                     src = { this.src }
                     alt="" />
             </if>
             <else>
                 <my-icon
                    ^
                     icon={ this.icon }
                     { ...this.undeclaredAttributes() }
                     class={ this._thumbnailClasses}
                     alt="" />
             </else>
         </g>;
     }
```

**Logs**
```
12:10:06 [ERROR] Unable to parse file: [...]/components/cell/CellThumbnail.jsx
12:10:06 [ERROR] Parse error at line 185 column 23:
12:10:06 
12:10:06 175: 
12:10:06 176:     render() {
12:10:06 177:         return <g>
12:10:06 178:             <if condition={ this.src }>
12:10:06 179:                 <img { ...this.undeclaredAttributes() }
12:10:06 180:                     class= { this._thumbnailClasses }
12:10:06 181:                     src = { this.src }
12:10:06 182:                     alt="" />
12:10:06 183:             </if>
12:10:06 184:             <else>
12:10:06 185:                 <my-icon
12:10:06                         ^
12:10:06 186:                     icon={ this.icon }
12:10:06 187:                     { ...this.undeclaredAttributes() }
12:10:06 188:                     class={ this._thumbnailClasses}
12:10:06 189:                     alt="" />
12:10:06 190:             </else>
12:10:06 191:         </g>;
12:10:06 192:     }
12:10:06 193: }
12:10:06 194: 
```

**Expected behavior**

File should be parsed without issue

--- a/javascript-frontend/src/main/java/org/sonar/javascript/parser/EcmaScriptLexer.java
+++ b/javascript-frontend/src/main/java/org/sonar/javascript/parser/EcmaScriptLexer.java
@@ -443,8 +443,8 @@ public enum EcmaScriptLexer implements GrammarRuleKey {
     b.rule(JSX_IDENTIFIER).is(SPACING, b.regexp(JavaScriptLexer.JSX_IDENTIFIER));
 
     // JSX looks at first letter: capital - JS identifier, small - html tag
-    // "this" is the exception of this rule
-    b.rule(JSX_HTML_TAG).is(SPACING, b.regexp("^(?!this)[a-z][\\w]*"));
+    // "this" is the exception to this rule
+    b.rule(JSX_HTML_TAG).is(SPACING, b.regexp("^(?!this)[a-z]\\w*+([-:.]\\w++)*+"));
 
     // Keywords
     b.rule(ASYNC).is(word(b, "async"));

--- a/javascript-frontend/src/test/java/org/sonar/javascript/tree/impl/expression/jsx/JsxElementTreeModelTest.java
+++ b/javascript-frontend/src/test/java/org/sonar/javascript/tree/impl/expression/jsx/JsxElementTreeModelTest.java
@@ -99,6 +99,14 @@ public class JsxElementTreeModelTest extends JavaScriptTreeModelTest {
   }
 
   @Test
+  public void complex_tag_name() throws Exception {
+    JsxStandardElementTree tree = parse("<foo:bar-baz.qux>Hello</foo:bar-baz.qux>", Kind.JSX_STANDARD_ELEMENT);
+    assertThat(tree.is(Kind.JSX_STANDARD_ELEMENT)).isTrue();
+    assertThat(expressionToString(tree.openingElement().elementName())).isEqualTo("foo:bar-baz.qux");
+    assertThat(expressionToString(tree.closingElement().elementName())).isEqualTo("foo:bar-baz.qux");
+  }
+
+  @Test
   public void short_fragment_syntax() throws Exception {
     JsxShortFragmentElementTree tree = parse("<> Hello <div/><div/></>", Kind.JSX_SHORT_FRAGMENT_ELEMENT);
 

